UI/Services/DistributedSessionService.cs

```diff
@@ -131,6 +136,84 @@ public async Task ClearAllAsync()
         _contextAccessor!.HttpContext!.Session!.Clear();
     }
 
+    public async Task<bool> IsDuplicateFormPostAsync(string formPostTimestampKey = "FormPostTimestamp")
+    {
+        var previousSubmissionTimestamp = await GetAsync<DateTimeOffset?>(formPostTimestampKey, FormPostPreKey);
+
+        var currentTimestamp = DateTimeOffset.UtcNow;
+
+        if (previousSubmissionTimestamp != null)
+        {
+            if ((currentTimestamp - previousSubmissionTimestamp.Value).TotalSeconds < IntegerConstants.SecondsClassifyAsDuplicateFormPost)
+            {
+                _logger.LogInformation("Multiple form POST (IsDuplicateFormPostAsync)");
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public async Task StartFormPostProcessingAsync(string formPostTimestampKey = "FormPostTimestamp")
+    {
+        var currentTimestamp = DateTimeOffset.UtcNow;
+        await SetAsync(formPostTimestampKey, currentTimestamp, FormPostPreKey);
+    }
+
+    public async Task SetFormPostResponseAsync<T>(T postResponseModel, string formPostModelKey = "FormPostModelKey")
+    {
+        await SetAsync(formPostModelKey, postResponseModel, FormPostPreKey);
+    }
+
+    public async Task<T> GetPreviousFormPostResponseAsync<T>(string formPostModelKey = "FormPostModelKey")
+    {
+        //Use Polly to retry 
+        int numberOfRetries = IntegerConstants.SecondsClassifyAsDuplicateFormPost;
```

Comment:
To avoid confusion, we should not reuse the constant value of `SecondsClassifyAsDuplicateFormPost` as the value for the variable name maxNumberOfRetries. Instead, we should choose a different value, such as `5`, to explicitly specify the maximum number of retries. So, the variable name should be maxNumberOfRetries and we can set its value as `5` if we want.

Reply:
I wasn't to sure what you were suggesting here.  Since we retry every second I think it should use the same int constant that we classify as a duplicate form post, otherwise there's a danger a different constants could be out of sync.  We should retry for as long as we classify the duplicate post is the intention.  I have updated the variable name to be more meaningful, to see if that helps.